sdk/communication/azure-communication-email/src/main/java/com/azure/communication/email/EmailClientBuilder.java

```diff
@@ -240,6 +263,15 @@ private AzureCommunicationServicesClientImpl buildInnerClient() {
         return client;
     }
 
+    private HttpPipelinePolicy createHttpPipelineAuthPolicy() {
+        if (this.tokenCredential != null) {
+            return new BearerTokenAuthenticationPolicy(
+                this.tokenCredential, "https://communication.azure.com//.default");
+        } else {
+            return new HmacAuthenticationPolicy(this.azureKeyCredential);
```

Comment:
What if this credential is also `null`? Do we check for that before calling this method?

Reply:
Good point, I'll throw an exception if this is the case.